realm/realm-library/src/main/java/io/realm/internal/OsObjectSchemaInfo.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal;
+
+
+import io.realm.RealmFieldType;
+
+/**
+ * Java wrapper for Object Store ObjectSchema.
+ *
+ * @see OsSchemaInfo
+ */
+public class OsObjectSchemaInfo implements NativeObject {
```

Comment:
This class is generally considered immutable right? Would it make sense to use the Builder pattern instead to create instances of it, then we only need to expose the getter methods afterwards?